pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -79,13 +80,26 @@
     rank_zero_warn,
 )
 from pytorch_lightning.utilities.exceptions import MisconfigurationException
-from pytorch_lightning.utilities.imports import _HOROVOD_AVAILABLE, _HPU_AVAILABLE, _IPU_AVAILABLE, _TPU_AVAILABLE
+from pytorch_lightning.utilities.imports import (
+    _HOROVOD_AVAILABLE,
+    _HPU_AVAILABLE,
+    _IPU_AVAILABLE,
+    _PYTHON_GREATER_EQUAL_3_8_0,
+    _TORCH_GREATER_EQUAL_1_11,
+    _TPU_AVAILABLE,
+)
 
 log = logging.getLogger(__name__)
 
 if _HOROVOD_AVAILABLE:
     import horovod.torch as hvd
 
+if _PYTHON_GREATER_EQUAL_3_8_0:
+    _LITERAL_WARN = Literal["warn"]
+else:
+    # there's an issue with the LightningCLI's help message using Literal on 3.7
```

Comment:
@mauvilsa looks like `Literal` does not work with the CLI and Python 3.7

https://github.com/PyTorchLightning/pytorch-lightning/runs/5843851902?check_suite_focus=true#step:17:3177

Reply:
`Literal` was introduced in python 3.8, see https://docs.python.org/3/library/typing.html#typing.Literal